source/ShipInfoPanel.cpp

```diff
@@ -468,32 +480,95 @@ void ShipInfoPanel::DrawWeapons(const Rectangle &bounds)
 	Color topColor;
 	bool hasTop = false;
 	auto layout = Layout(static_cast<int>(LABEL_WIDTH), Truncate::BACK);
+
+	// First calculate how many pages are needed. One is always there.
+	pages = 1;
+	int gunIndex = 0;
+	int turretIndex = 0;
+	int rowsPerPage = static_cast<int>((sprite->Height() * scale) / 10.) - 1;
```

Comment:
Again, `sprite->Height() * scale` may actually be muych less than the actual available height.